src/storage/cloud_storage.rs

```diff
@@ -73,6 +73,23 @@ fn item_to_metadata(item: Item) -> ObjectMetadata {
     }
 }
 
+fn item_to_file_info(item: Item) -> Fileinfo<PathBuf, ObjectMetadata> {
+    Fileinfo {
+        path: PathBuf::from(item.name),
+        metadata: ObjectMetadata {
+            last_updated: match u64::try_from(item.updated.timestamp_millis()) {
+                Ok(timestamp) => SystemTime::UNIX_EPOCH.checked_add(Duration::from_millis(timestamp)),
+                _ => None,
+            },
```

Comment:
Didn't test this, but I think you can go from Result to Option with `ok()` which discards the error:

```rust
u64::try_from(item.updated.timestamp_millis())
  .map(|timestamp| SystemTime::UNIX_EPOCH.checked_add(Duration::from_millis(timestamp)))
  .ok()
```

https://doc.rust-lang.org/std/result/enum.Result.html#method.ok

Reply:
with this we have an Option<Option<u64>> and flatten() is unstable.